tensorflow/core/kernels/as_string_op.cc

```diff
@@ -92,6 +92,16 @@ class AsStringOp : public OpKernel {
     if (width_ <= -1) {
       width_ = 0;
     }
+    
+    // Check for unreasonably large width values to prevent memory exhaustion
+    // and segmentation faults when absl::StrFormat tries to allocate buffers
+    const int kMaxReasonableWidth = 1000000;  // 1 million characters
```

Comment:
Allocation with large numbers fails and returns a null pointer. Better to check for that than to use an arbitrary large constant which has no justification.

See also #105318 comments